Rewrite HttpFaultInjector to use ILogger and built-in ASP.NET Core logging

lmolkova opened the pull request:
- Logging:
  - Leverages built-in ASP.NET Core & HttpClientFactory logging
  - Uses ILogger for a few remaining messages not covered by default logging
  - Adds OTel Console exporter to print correlated and structured logs to console
  - Configures logging to capture headers and other info captured before
- Configuration:
  - Updates default endpoints and allows to leverage default kestrel config via env vars/cmd line args
- Misc refactorings


Example:
```
LogRecord.Timestamp:               2023-12-19T21:46:21.1159635Z
LogRecord.TraceId:                 6728003b59ab450cb730ab8c5670f3f9
LogRecord.SpanId:                  5f82c7d70faf07d6
LogRecord.TraceFlags:              None
LogRecord.CategoryName:            Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware
LogRecord.Severity:                Info
LogRecord.SeverityText:            Information
LogRecord.FormattedMessage:        Request:
Protocol: HTTP/1.1
Method: GET
Scheme: http
PathBase:
Path: /path/to/something
Accept: */*
Host: localhost:7777
User-Agent: curl/8.4.0
X-Upstream-Base-Uri: https://example.com
x-ms-faultinjector-response-option: f
LogRecord.Body:                    Request:
Protocol: HTTP/1.1
Method: GET
Scheme: http
PathBase:
Path: /path/to/something
Accept: */*
Host: localhost:7777
User-Agent: curl/8.4.0
X-Upstream-Base-Uri: https://example.com
x-ms-faultinjector-response-option: f
LogRecord.Attributes (Key:Value):
    Protocol: HTTP/1.1
    Method: GET
    Scheme: http
    PathBase:
    Path: /path/to/something
    Accept: */*
    Host: localhost:7777
    User-Agent: curl/8.4.0
    X-Upstream-Base-Uri: https://example.com
    x-ms-faultinjector-response-option: f
LogRecord.EventId:                 1
LogRecord.EventName:               RequestLog
```

mikeharder commented on tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/appsettings.json:
```diff
@@ -0,0 +1,10 @@
+      "Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware": "Information"
+    }
+  }
+}
```
```suggestion
}

```

mikeharder reviewed (approved):
Looks great.  Thank you!!!

lmolkova commented:
@mikeharder I've rebased this change on top of streaming and changed controller to middleware (thanks for the suggestion earlier).

LMK if you want to take another too.
I've done some testing locally and going to run storage tests against this version to check it.

mikeharder commented on tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs:
```diff
@@ -71,364 +30,59 @@ public static void Main(string[] args)
+            });
 
-            Log($"StatusCode: {upstreamResponse.StatusCode}");
+            builder.Services.AddControllers();
```
Do we still need this if we don't have any controllers?  It's probably a no-op so doesn't matter, but curious about best practices.

lmolkova replied:
good catch! no we don't need it anymore

mikeharder commented:
@lmolkova: I reviewed this again and it LGTM.
